pkg/controllers/workgenerator/controller_test.go

```diff
@@ -119,3 +124,139 @@ func Test_getWorkNameFromSnapshotName(t *testing.T) {
+						Generation: 12,
+					},
+					Status: workv1alpha1.WorkStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:               workapi.ConditionTypeApplied,
+								Status:             metav1.ConditionTrue,
+								ObservedGeneration: 12,
+							},
+						},
+					},
+				},
+			},
+			generation: 1,
+			want: metav1.Condition{
+				Status:             metav1.ConditionTrue,
+				Type:               string(fleetv1beta1.ResourceBindingApplied),
+				Reason:             allWorkAppliedReason,
+				ObservedGeneration: 1,
+			},
+		},
+		"applied should be false if not all work applied to the latest generation": {
+			works: map[string]*workv1alpha1.Work{
+				"notAppliedWork1": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "work1",
+						Generation: 123,
+					},
+					Status: workv1alpha1.WorkStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:               workapi.ConditionTypeApplied,
+								Status:             metav1.ConditionTrue,
+								ObservedGeneration: 122, // not the latest generation
+							},
+						},
+					},
+				},
+				"appliedWork2": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "work2",
+						Generation: 12,
+					},
+					Status: workv1alpha1.WorkStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:               workapi.ConditionTypeApplied,
+								Status:             metav1.ConditionTrue,
+								ObservedGeneration: 12,
+							},
+						},
+					},
+				},
+			},
+			generation: 1,
+			want: metav1.Condition{
+				Status:             metav1.ConditionFalse,
+				Type:               string(fleetv1beta1.ResourceBindingApplied),
+				Reason:             workNotAppliedReason,
+				ObservedGeneration: 1,
+			},
+		},
+		"applied should be false if not all work has applied": {
+			works: map[string]*workv1alpha1.Work{
+				"appliedWork1": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "work1",
+						Generation: 123,
+					},
+					Status: workv1alpha1.WorkStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:               workapi.ConditionTypeApplied,
+								Status:             metav1.ConditionTrue,
+								ObservedGeneration: 122, // not the latest generation
+							},
+						},
+					},
+				},
+				"notAppliedWork2": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "work2",
+						Generation: 12,
+					},
+				},
+			},
+			generation: 1,
+			want: metav1.Condition{
+				Status:             metav1.ConditionFalse,
+				Type:               string(fleetv1beta1.ResourceBindingApplied),
+				Reason:             workNotAppliedReason,
+				ObservedGeneration: 1,
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			binding := &fleetv1beta1.ClusterResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test",
+					Generation: tt.generation,
+				},
+			}
+			if got := buildAllWorkAppliedCondition(tt.works, binding); !reflect.DeepEqual(got, tt.want) {
```

Comment:
```suggestion
			if got := buildAllWorkAppliedCondition(tt.works, binding); !cmp.Equal(got, tt.want) {
```

prefer to use cmp.Equal
https://github.com/golang/go/wiki/TestComments#equality-comparison-and-diffs
`
Prefer cmp for new code, and consider updating older code to use cmp where practical. reflect.DeepEqual is sensitive to changes in unexported fields and other implementation details.`